example/flaskr/auth.py

```diff
@@ -74,9 +73,16 @@ def register():
                 (username, generate_password_hash(password)),
             )
             db.commit()
-            return redirect(url_for("auth.login"))
 
-        flash(error)
+            # create successful registration response object
+            response = json.dumps({
+                "category": "info",
+                "message": "User account successfully registered!"
+            })
+
+            return redirect(url_for("auth.login", response=response))
```

Comment:
This is sort of the only thing I'm disappointed in as there is no way to determine if the user was successfully redirected to login or simply went to the page themselves. So this was the only solution which you can see in the test is kind of ugly but it works.

Reply:
I was able to query the response object for status. Is a status of '302 -FOUND' sufficient to indicate a successful redirect? I left a suggestion over in `test_auth.py`
Also, if we're going to be using a lot of URL strings in code, we might want to see if the python [`requests`](https://requests.readthedocs.io/en/master/) plays well with Flask. I think it does a good job of encoding and passing URL strings with all the correct escape sequences.

Reply:
Yeah, 302 is a permanent redirect so that should be fine unless POSTs follow redirects? Although I don't think we have any of those we need to worry about currently at least.
I think using `requests` might be more cleaner if we can send POST params (i.e. in the request payload instead of in the URL). I just didn't want to waste too much time swapping everything out since we're so close to the end of the sprint.